lib/buffer.js

```diff
@@ -64,7 +64,17 @@ function alignPool() {
  * much breakage at this time. It's not likely that the Buffer constructors
  * would ever actually be removed.
  **/
+var newBufferWarned = false;
 function Buffer(arg, encodingOrOffset, length) {
+  if (!new.target && !newBufferWarned) {
+    newBufferWarned = true;
+    process.emitWarning(
+      'Using Buffer without `new` will soon stop working. ' +
+      'Use `new Buffer`, or preferably ' +
+      'Buffer.from, Buffer.allocUnsafe or Buffer.alloc instead.',
```

Comment:
I would keep all three listed and just drop the 'Use new Buffer'... just point to the three new constructors in the order: `Buffer.from()`, `Buffer.alloc()`, and `Buffer.from()`.

Also, a small nit: @seishun, can you add the `()` after each of the functions?